### ISSUE TITLE
Questions about the Dynamic Scheduling Module of ABC Function

### ISSUE THREAD

**Issue by zhouyaguang** [WWFUG/Flexible-Job-Shop-Scheduling-Problem-with-Variable-Processing-Time#1]
Hello, I saw in the picture you pasted that there are dynamic scheduling values for delta=0.1 to 0.5, but when I actually ran the ABC function, I found that its dynamic scheduling module was not completed. Has the program been uploaded completely? thanks

**Comment by WWFUG** [comment]
Hi, the dynamic scheduling is implemented as the method AbcScheduler::dynamic_optimize().
In the main.py, comment out Line 33 and uncomment Line 34 should fix it.

**Comment by zhouyaguang** [comment]
Thanks for your reply, actually I have tried to fix some issue encountered in your code. As you written in the ABC class, there are two bugs in the dunamic_optimize funtion, which mainly casues list range out of index problem when dynamic optimizing it. So am I misunderstand the code, or there are still some bugs need to be fixed in this code? Thanks again for your reply, I really interested in your code.

**Comment by zhouyaguang** [comment]
Actually you can check out the incor_proc_jop_MA and incor_proc_jop_OS function, they mainly deal with the dynamic optimization problem, but they are not completed. When debug them, I found that these two functions didin't realize dynamic remove or insert operations as we expected. And the inital parameters like machine_to_started_jop and job_to_unprocessed_op is meanwhile the list of the dynamic optimization function, they are totaly not used during dynamic optimizing. Maybe I misunderstood your code, but I would appreciate it very much if you could check out your code. Thanks.

**Comment by WWFUG** [comment]
Can you describe what you expect these two procedures to be?
For OS, the incor_proc_jop_OS will check the already started operations, put them to the prefix_OS, remove the entry in the original OS, and then concatenate the prefix_OS and the original OS together. It should be complete.
For MA, the incor_proc_jop_MA will fix the machine entries in MA of those already started operations to their assigned machines.
machine_to_started_jop is used in both incor_proc_* functions and job_to_unprocessed_op is used in incor_proc_job_OS.

**Comment by zhouyaguang** [comment]
Yeah, I totally understand these two functions. The questions are mainly two parts. One is that when I just run the s.dynamic_optimize(100, 20, time_delta), there will be errir in incor_proc_jop_OS, which is ValueError: list.remove(x): x not in list. Another one is when I debug these two functions, I found these two functions didn't remove index of operations when we expected. You can check out in the sentence for _ in range(del_count): OS.remove(job+1), because del_count is always 0, so remove function didn't remove anything, and the dynamic process actually didn't exists. I draw above conclusion depend on debuging process, and I analyze if you fix out these bugs while the latest version is still not uploaded. Because I saw in your code there are two bugs in the dynamic optimization module, and maybe you forget to upload after you fix it.
